contessa/consistency_checker.py

```diff
@@ -0,0 +1,136 @@
+    """
+
+    COUNT = "count"
+    DIFF = "difference"
+
+    def __init__(self, left_conn_uri_or_engine, right_conn_uri_or_engine=None):
+        self.left_conn_uri_or_engine = left_conn_uri_or_engine
+        self.left_conn = Connector(left_conn_uri_or_engine)
+        if right_conn_uri_or_engine is None:
+            self.right_conn_uri_or_engine = self.left_conn_uri_or_engine
+            self.right_conn = self.left_conn
+        else:
+            self.right_conn_uri_or_engine = right_conn_uri_or_engine
+            self.right_conn = Connector(right_conn_uri_or_engine)
+
+    def run(
+        self,
+        check_type: str,
+        left_check_table: Dict,
+        right_check_table: Dict,
+        result_table: Dict,
+        context: Optional[Dict] = None,
+    ):
+        left_check_table = Table(**left_check_table)
+        right_check_table = Table(**right_check_table)
+        result_table = ResultTable(**result_table)
+        context = self.get_context(left_check_table, right_check_table, context)
+
+        result = self.do_consistency_check(
+            check_type, left_check_table, right_check_table, context
+        )
+
+        quality_check_class = create_default_check_class(
+            result_table, check_type="consistency"
+        )
+        self.ensure_table(quality_check_class)
+        self.insert(quality_check_class, result)
+
+    @staticmethod
+    def get_context(
+        left_check_table: Table,
+        right_check_table: Table,
+        context: Optional[Dict] = None,
+    ) -> Dict:
+        """
+        Construct context to pass to executors. User context overrides defaults.
+        """
+        ctx_defaults = {
+            "left_table_fullname": left_check_table.fullname,
+            "right_table_fullname": right_check_table.fullname,
+            "task_ts": datetime.now(),
+        }
+        ctx_defaults.update(context)
+        return ctx_defaults
+
+    def do_consistency_check(
+        self,
+        check_type: str,
+        left_check_table: Table,
+        right_check_table: Table,
+        context: Dict = None,
+    ):
+        """
+        Run quality check for all rules. Use `qc_cls` to construct objects that will be inserted
+        afterwards.
+        """
+        left_result = self.run_query(
+            left_check_table.fullname, self.left_conn, check_type
+        )
+        right_result = self.run_query(
+            right_check_table.fullname, self.right_conn, check_type
+        )
+        return {
+            "check": {"name": check_type, "description": ""},
+            "status": left_result == right_result,
+            "left_table_name": left_check_table.fullname,
+            "right_table_name": right_check_table.fullname,
+            "context": context,
+        }
+
+    def run_query(self, table_name: str, conn: Connector, type: str):
+        if type == self.COUNT:
+            column = "count(*)"
+        else:
+            column = "*"
```

Comment:
I believe this won't work in case columns are in different order - it shouldn't happen if its raw copy of a table, but it can. Maybe we should put here column names instead of the `*` and sort them. what do you think? 

Also, maybe here the db.py would be helpful - https://github.com/yhat/db.py

Reply:
Indeed this wont work in such case. I added a test that tests for this to not work correctly with a TODO and I suggest we introduce db.py in a separate PR as this one would get polluted with that change. Then we can fix this issue. Would you agree?

Reply:
Agree on db.py, dont agree on fixing the issue. what I think we should do (pick one of):

1. Fix the issue now (without db.py), refactor contessa to db.py later (also with the feature)
2. Remove feature now, refactor contessa to db.py, implement new diff feature (with db.py)

I don't think it is responsible from us to keep there something that doesn't work and we even know about it.

Reply:
I dont see it so radically, the feature works as it is, it only doesnt work for a scenario where compared tables have same columns and same data but columns are in different order - which should actually be more of an edge case, not really a typical situation. 

Anyway, I did 1. now as it was just a matter of getting the column names and listing them sorted.

Reply:
thanks